From be0a91c240982568fbce2736a971c3614c025dd0 Mon Sep 17 00:00:00 2001
From: Veena Sri Vallala <vsri@qti.qualcomm.com>
Date: Mon, 21 Apr 2025 20:41:19 +0530
Subject: [PATCH 7/8] rtl8261: Enable Clause 45 protocol

Set is_c45 flag in rtl8261 driver to enable
clause 45 based address access.

Signed-off-by: Veena Sri Vallala <vsri@qti.qualcomm.com>
---
 linux-5.10.146/drivers/net/phy/rtk/rtk_phy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/linux-5.10.146/drivers/net/phy/rtk/rtk_phy.c b/linux-5.10.146/drivers/net/phy/rtk/rtk_phy.c
index bc02b09..054101b 100644
--- a/linux-5.10.146/drivers/net/phy/rtk/rtk_phy.c
+++ b/linux-5.10.146/drivers/net/phy/rtk/rtk_phy.c
@@ -82,6 +82,7 @@ static int rtl826xb_probe(struct phy_device *phydev)
     struct rtk_phy_priv *priv = NULL;
     int data = 0;
 
+    phydev->is_c45 = true;
     priv = devm_kzalloc(&phydev->mdio.dev, sizeof(struct rtk_phy_priv), GFP_KERNEL);
     if (!priv)
     {
-- 
2.34.1

